docs(hybrid-mode) add 2.5 related version compatibility behavior change (#2973)
Kong/docs.konghq.com: app/gateway-oss/2.5.x/hybrid-mode.md

### Version compatibility

{{site.ce_product_name}} control planes only allow connections from data planes
with the same major version.
Control planes won't allow connections from data planes with newer minor versions.

> For example, a {{site.ce_product_name}} v2.4.2 control plane:

Reviewer: I think maybe we want to "rebase" the version examples to 2.5 here. Since although the
logic is correct, in realty a 2.4.2 Kong won't have those behaviour though.

Author: Changed.

- Accepts a {{site.ce_product_name}} 2.4.0, 2.4.1 and 2.4.2 data plane
- Accepts a {{site.ce_product_name}} 2.3.8, 2.2.1 and 2.2.0 data plane

> - Accepts a {{site.ce_product_name}} 2.4.3 data plane

Reviewer: ```suggestion
- Accepts a {{site.ce_product_name}} 2.4.3 data plane (patch version on data plane is newer)
```

- Rejects a {{site.ce_product_name}} 1.0.0 data plane (major version differs)
- Rejects a {{site.ce_product_name}} 2.5.0 data plane (minor version on data plane is newer)

Furthermore, for every plugin that is configured on the {{site.ce_product_name}}
control planes, new configs are only pushed to data planes that have those configured
plugins installed and loaded, and the major version of those configured plugins must
be the same on both control planes and data planes, and minor version on the data planes

> could not be newer than versions installed on the control planes. (Configured plugins means

Reviewer: Let's also mention patch version is ignored regardless or its value.

Author: Added.

### Version compatibility

{{site.ce_product_name}} control planes only allow connections from data planes
with the same major version.
Control planes won't allow connections from data planes with newer minor versions.

For example, a {{site.ce_product_name}} v2.5.2 control plane:

- Accepts a {{site.ce_product_name}} 2.5.0, 2.5.1 and 2.5.2 data plane
- Accepts a {{site.ce_product_name}} 2.3.8, 2.2.1 and 2.2.0 data plane
- Accepts a {{site.ce_product_name}} 2.5.3 data plane (newer patch version on the data plane is accepted)
- Rejects a {{site.ce_product_name}} 1.0.0 data plane (major version differs)

> - Rejects a {{site.ce_product_name}} 2.5.0 data plane (minor version on data plane is newer)

Author: That's a good catch. I think it's okay to mention `2.6.0` as it is for illustrative purpose only... That way the `For example, a {{site.ce_product_name}} v2.5.2 control plane:` sentence above actually makes more sense for the current release as well.

> - Rejects a {{site.ce_product_name}} 2.5.0 data plane (minor version on data plane is newer)

Author: Changed to `2.6.0`.

Furthermore, for every plugin that is configured on the {{site.ce_product_name}}

> control planes, new configs are only pushed to data planes that have those configured

Reviewer 2: ```suggestion
control plane, new configs are only pushed to data planes that have those configured
```

> plugins installed and loaded, and the major version of those configured plugins must

Reviewer 2: ```suggestion
plugins installed and loaded. The major version of those configured plugins must
```

> be the same on both control planes and data planes, and minor version on the data planes

Reviewer 2: ```suggestion
be the same on both the control planes and data planes. Also, the minor versions of the plugins on the data planes
```

could not be newer than versions installed on the control planes. Note that similar to
{{site.ce_product_name}} version checks, plugin patch versions are also ignored

> when determining the compatibility. (Configured plugins means

Reviewer 2: ```suggestion
when determining the compatibility. 

{:.important}
```

> any plugin that is either enabled globally, or configured via Services/Routes/Consumers.)

Reviewer 2: ```suggestion
> Configured plugins means any plugin that is either enabled globally or configured by Services, Routes, or Consumers.
```

For example, if a {{site.ce_product_name}} control plane has `plugin1` v1.1.1

> and `plugin2` v2.1.0 installed, and `plugin1` is configured via a `Route` object:

Reviewer 2: ```suggestion
and `plugin2` v2.1.0 installed, and `plugin1` is configured by a `Route` object:
```

- It accepts {{site.ce_product_name}} data planes with `plugin1` v1.1.2,

> `plugin2` not installed

Reviewer 2: ```suggestion
and `plugin2` not installed.
```

- It accepts {{site.ce_product_name}} data planes with `plugin1` v1.1.2,

> `plugin2` v2.1.0 and  `plugin3` v9.8.1 installed

Reviewer 2: ```suggestion
`plugin2` v2.1.0, and  `plugin3` v9.8.1 installed.
```

- It rejects {{site.ce_product_name}} data planes with `plugin1` v1.2.0,

> `plugin2` v2.1.0 installed (minor version of plugin on data plane is newer)

Reviewer 2: ```suggestion
 and `plugin2` v2.1.0 installed (minor version of plugin on data plane is newer).
```

- It accepts {{site.ce_product_name}} data planes with `plugin1` v1.1.1,

> `plugin3` v9.8.1 installed

Reviewer 2: ```suggestion
 and `plugin3` v9.8.1 installed.
```

- It rejects {{site.ce_product_name}} data planes with `plugin1` not installed

> (plugin configured but not installed on data plane)

Reviewer 2: ```suggestion
(plugin configured on control plane but not installed on data plane).
```

If the compatibility checks fail, the control plane stops

> pushing out new config to the incompatible data planes to avoid breaking them.

Reviewer 2: ```suggestion
pushing out new configs to the incompatible data planes to avoid breaking them.
```

> When new configs are sent out from control planes to data planes, the data planes

Reviewer 2: ```suggestion
When new configs are sent from control planes to data planes, the data planes
```

> runs it's own schema validator against control plane supplied DB-less config. Data planes

Reviewer 2: ```suggestion
run their own schema validator against a control plane supplied DB-less config. Data planes
```

> rejects any config that could not pass the schema validation.

Reviewer 2: ```suggestion
reject any config that does not pass the schema validation.
```

One exception to the validation on data planes is that if the value of a control plane

> supplied field is `null` and data plane's schema does not have information about

Reviewer 2: ```suggestion
supplied field is `null` and the data plane's schema does not have information about
```
